config/config.go

```diff
@@ -257,10 +268,10 @@ func initConsensusProtocols() {
 
 	// Base consensus protocol version, v7.
 	v7 := ConsensusParams{
-		UpgradeVoteRounds:   10000,
-		UpgradeThreshold:    9000,
-		UpgradeWaitRounds:   10000,
-		MaxVersionStringLen: 64,
+		UpgradeVoteRounds:        10000,
+		UpgradeThreshold:         9000,
+		DefaultUpgradeWaitRounds: 10000,
```

Comment:
I think you want to specify `MinUpgradeWaitRounds` and `MaxUpgradeWaitRounds` here.

Reply:
I'm having a value of 0 denote `DefaultUpgradeWaitRounds`, which might be a bit confusing.

Reply:
(See https://github.com/algorand/go-algorand/pull/650/files#diff-590486bd16364da6743388800e2ea504R312)

Reply:
Ah, I see, setting them to zero forces the block to have a zero value which is backwards-compatible..

It would be good to revise the description of these fields in the config type definition above, to make it clear that these apply to the block header representation, rather than the actual number of rounds to wait before upgrading.

Reply:
Sounds good.

The other thing I considered was to have the delay implicit in `NextProtocolSwitchOn`, so no field needs to be introduced in the block header. The drawback of this is that it makes `applyUpgradeVote` a bit messier (see the first commit on this PR). For consistency, I've decided to add the field to make it explicit.